### ISSUE TITLE
Improve License Information in manifest

### ISSUE THREAD

**Issue by swaldhoer** [microsoft/winget-pkgs#855]
# Description of the new feature/enhancement

Projects that are hosted on some kind of services that allows browsing the sources should include the exact license information not only the generic link to the project license.

# Proposed technical implementation details (optional)

Where possible the versions exact license file should be linked, as license may (and they do) change during a project.

I think adding a entry to the *.yaml would solve this issue:

Example for the [git package](https://github.com/microsoft/winget-pkgs/blob/a20c7e0e427c5cf76d1f494e44cc177911bcee26/manifests/Git/Git/2.26.2.yaml):

The project has an accessible repository for its releases (not only a zip release). We add a key ``ExactLicenseUrl`` and the manifest would look like this:

```yaml
# ...
Homepage: https://git-scm.com/
License: GNU General Public License, version 2
ExactLicenseUrl: https://github.com/git-for-windows/git/blob/v2.26.2.windows.1/COPYING
LicenseUrl: https://github.com/git-for-windows/git/blob/master/COPYING
Description: Git version control system.
InstallerType: Inno
# ...
```

And now you can see general license information which might be interesting, but the more important thing, you now know the license of the version you are currently using.

**Comment by jamierocks** [comment]
Given the manifests are for specific versions, I think its reasonable to use the exact URL for manifests - without the need for another key.

```yaml
# ...
Homepage: https://git-scm.com/
License: GNU General Public License, version 2
LicenseUrl: https://github.com/git-for-windows/git/blob/v2.26.2.windows.1/COPYING
Description: Git version control system.
InstallerType: Inno
# ...
```

**Comment by swaldhoer** [comment]
You are right, that it could simply be fixed by using the exact version in every manifest with the ``LicenseUrl`` property, but as a lot of packages (GitHub (70), GitLab (4) and SourceForge (1)) link to the master-branch license file I assumed that this is desired behavior by the developers and therefore adding a new key was the solution to me. There are a two that link to specific version, e.g. [Go](https://github.com/microsoft/winget-pkgs/blob/a20c7e0e427c5cf76d1f494e44cc177911bcee26/manifests/Golang/Go/1.14.3.yaml#L6), GitHub Desktop in version [2.5.0](https://github.com/microsoft/winget-pkgs/blob/a20c7e0e427c5cf76d1f494e44cc177911bcee26/manifests/GitHub/GitHubDesktop/2.5.0.yaml#L6) (while its the generic link in version [2.4.3](https://github.com/microsoft/winget-pkgs/blob/a20c7e0e427c5cf76d1f494e44cc177911bcee26/manifests/GitHub/GitHubDesktop/2.4.3.yaml#L7)). So there seems no general rule. I personally think, linking to the exact version file is the correct way to do it.
 
At least this should somehow be a consistent and enforced by the reviewers in form of a rule like :

1. If possible, add a link to the license file of this exact version.
1. If this is not possible, link to the generic version of the license.

As of a20c7e0e427c5cf76d1f494e44cc177911bcee26 there are 330 manifests in this repository, so 75 manifest are a big part that is affected by that.

**Comment by denelon** [comment]
We agree that the best "LicenseURL" value is one that is associated with the specific version of a package. As we look deeper into the automation of generating manifests from GitHub releases this will be taken into consideration. Until then, it will likely continue to be a community "best effort" to provide the best license values.

**Comment by Croydon** [comment]
I think for the annotation of the licenses SPDX license identifiers should be adopted

https://spdx.org/licenses/

**Comment by Croydon** [comment]
Hi! This is my generic reply for every issue which is still relevant and is on the edge of getting closed by a stale bot.

____________________________

I think this bot is way too aggressive..

**Comment by Croydon** [comment]
Hi! This is my generic reply for every issue which is still relevant and is on the edge of getting closed by a stale bot.

**Comment by denelon** [comment]
The message from the bot wasn't correct. The issue will not be auto closed unless it has the "Needs-author-feedback" and hasn't had any activity for 7 days after the "Status: no recent activity" label has been added. This will not be auto closed based on the current bot rules.

**Comment by denelon** [comment]
We also have #903 in the backlog.

**Comment by denelon** [comment]
@Croydon we've added several new fields since this issue was created. I think we're still having challenges with links to licenses for open projects pointing to trunk branches, but this seems to happen less frequently now. I took a look at the SPDX licensing, but those only apply to "open source" licenses. That would prevent us from using an enumeration to validate the "License" field. I've also seen examples where source code is one license and a compiled package is covered under a different license.

I think we could write a "best practice" for using the SPDX Identifiers. Thoughts?
